### ISSUE TITLE
Error TS1005 = expected when transpiling

### ISSUE THREAD

**Issue by antonybudianto** [angular/vladivostok#39]
When I tried to transpile the code, I got error as following
```bash
/node_modules/@angular/router/directives/router_outlet.d.ts(10,14): error TS1005: '=' expected.
```

I used latest **gulp-typescript**, which have **typescript** version `1.8.10`,
I believe it got to do with TypeScript versioning issue because of these offending lines `directives/router_outlet.d.ts(10,14)`
```ts
readonly isActivated: boolean;
readonly component: Object;
readonly activatedRoute: ActivatedRoute;
```
looks like the generated definition files is not compatible, I see you used the dev version of typescript https://github.com/angular/vladivostok/blob/master/package.json#L66

Angular 2 version: RC 1, 2.0.0-rc.1
Router: 3.0.0-alpha.3

**Comment by jeffschwartz** [comment]
I am also getting the error as well as others as the following shows:

~/Development/preamble-ts/preamble-ts-docs$ tsc
node_modules/@angular/router/directives/router_outlet.d.ts(10,14): error TS1005: '=' expected.
node_modules/@angular/router/directives/router_outlet.d.ts(10,25): error TS1005: ';' expected.
node_modules/@angular/router/directives/router_outlet.d.ts(11,14): error TS1005: '=' expected.
node_modules/@angular/router/directives/router_outlet.d.ts(11,23): error TS1005: ';' expected.
node_modules/@angular/router/directives/router_outlet.d.ts(12,14): error TS1005: '=' expected.
node_modules/@angular/router/directives/router_outlet.d.ts(12,28): error TS1005: ';' expected.
node_modules/@angular/router/router.d.ts(60,14): error TS1005: '=' expected.
node_modules/@angular/router/router.d.ts(60,25): error TS1005: ';' expected.
node_modules/@angular/router/router.d.ts(61,14): error TS1005: '=' expected.
node_modules/@angular/router/router.d.ts(61,21): error TS1005: ';' expected.
node_modules/@angular/router/router.d.ts(62,14): error TS1005: '=' expected.
node_modules/@angular/router/router.d.ts(62,20): error TS1005: ';' expected.
node_modules/@angular/router/router.d.ts(62,39): error TS1005: '(' expected.
node_modules/@angular/router/router_state.d.ts(29,27): error TS1110: Type expected.
node_modules/@angular/router/router_state.d.ts(31,123): error TS1110: Type expected.
node_modules/@angular/router/router_state.d.ts(35,24): error TS1110: Type expected.
node_modules/@angular/router/router_state.d.ts(36,97): error TS1110: Type expected.
node_modules/@angular/router/url_tree.d.ts(7,24): error TS1110: Type expected.
node_modules/@angular/router/url_tree.d.ts(10,27): error TS1110: Type expected.
node_modules/@angular/router/utils/tree.d.ts(4,14): error TS1005: '=' expected.
node_modules/@angular/router/utils/tree.d.ts(4,18): error TS1005: ';' expected.
node_modules/@angular/router/utils/tree.d.ts(5,23): error TS1110: Type expected.
node_modules/@angular/router/utils/tree.d.ts(7,27): error TS1110: Type expected.

I am using typescript Version 1.8.10

**Comment by brandonroberts** [comment]
You should be using the latest version of TypeScript. Look at the
package.json for the correct TypeScript version you need.

**Comment by jeffschwartz** [comment]
@brandonroberts thanks but that would break atom's typescript tooling via the atom typescript plugin which is built using typescript v1.8.10. Why not just use const instead of readonly which would make the router usable by many more testers?

**Comment by antonybudianto** [comment]
sorry, the latest should be `1.8.10` right? not the nightly build

**Comment by brandonroberts** [comment]
Yes, the latest stable is 1.8.10. The latest version of TypeScript doesn't
prevent you from using the atom TypeScript plugin, it just doesn't
recognize those new features yet.

**Comment by jeffschwartz** [comment]
@brandonroberts sorry but that makes no sense at all. The router depends on a version of typescript that is ahead of the release version. The atom plugin is based off of the v1.10.8 of typescript. Therefore trying to use this router with atom and the atom typescript plugin will produce a failed transpilation due to the unrecognized and yet to be supported typescript features the router is using. Also, adding node_modules to the exclude property in tsconfig.json will not fix this issue as once a non excluded file references a module in node_modules it will no longer be excluded. This is a lose lose proposition for those of us who were looking to be early adopters of this router.

**Comment by jeffschwartz** [comment]
I just don't understand why the router was developed using edge compiler features. Couldn't v1.8.10 had been used and "const" have been used instead of "reasonly" and transitioned to the new features as they were added to released versions? I believe poor choices were made that now negatively impacts early adoption. Another misstep for router imho.

**Comment by jeffschwartz** [comment]
I just filed #40

**Comment by born2net** [comment]
you can also install for 
``` npm install typescript@next``` for latest

**Comment by jeffschwartz** [comment]
@born2net that doesn't address the incompatibility issue. Secondly, not everyone is free to change the version of the compiler for the projects they are working on. Many of us are restricted to use the stable release version and are unable to use edge versions of the compiler.

**Comment by Levia-** [comment]
Personally, I'm encountering this in Visual Studio with the new angular router. VS doesn't allow you to change the typescript compiler version easily.

**Comment by ribizli** [comment]
@Levia- You can change the VSCode TypeScript version in the config:
```
    // Specifies the folder path containing the tsserver and lib*.d.ts files to use.
    "typescript.tsdk": null,
```
(Install `typescript@next` globally and reference the path)

**BUT**: I've installed the `@next` version for my project and now I'm getting an out of memory error during compiling. (It compiles fine with `@1.8.10` except the router errors). So we introduce another issue by using a not released version of typescript!
```
FATAL ERROR: CALL_AND_RETRY_LAST Allocation failed - process out of memory
```

**Comment by Levia-** [comment]
I've since been able to resolve all issues by using installing Nodejs manually, using gulp-typescript and typescript@next, turning off VS' typescript compilation and doing everything through gulp.

(You can use typescript@next with gulp-typescript as shown here: https://www.npmjs.com/package/gulp-typescript under the heading TypeScript Version)

**Comment by ribizli** [comment]
If I use the TypeScript version mentioned in the router's `package.json` (`@1.9.0-dev.20160409`) the memory problem goes away. (Seems there is a broken feature in the last few nightlies)

**Comment by deastr** [comment]
I'm using VS.NET 2015 and I've installed TypeScript 1.9.0-dev.20160409 using [VSDevMode.ps1](https://github.com/Microsoft/TypeScript/wiki/Nightly-drops#visual-studio-2013-and-2015). It did get rid some of the compile errors but not all. Is still get some 
> Build: '=' expected. node_modules\@angular\router\directives\router_outlet.d.ts	10

and 

> Build: ';' expected. node_modules\@angular\router\directives\router_outlet.d.ts	10

errors. 

I've also tried 1.9.0-dev.20160525 but didn't do any different than 1.9.0-dev.20160409.

Any ideas?
